controllers/v1/solutions.js

```diff
@@ -16,64 +16,1066 @@ module.exports = class  Solutions extends Abstract{
+    }
+
+
+
+     /**
+    * @api {post} /project/v1/solutions/targetedSolutions?type=:solutionType&page=:page&limit=:limit&search=:search&filter=:filter
+    * List of assigned solutions and targetted ones.
+    * @apiVersion 1.0.0
+    * @apiGroup Solutions
+    * @apiSampleRequest /project/v1/solutions/targetedSolutions?type=observation&page=1&limit=10&search=a&filter=assignedToMe
+    * @apiParamExample {json} Request:
+    * {
+    *   "entityType" : "block",
+        "entities" : [
+            "5fd1b52ab53a6416aaeefc80",
+            "5fd098e2e049735a86b748ac"
+        ],
+        "roles" : "BEO"
+    }
+    * @apiParamExample {json} Response:
+    {
+    "message": "Solutions fetched successfully",
+    "status": 200,
+    "result": {
+        "data": [
+            {
+                "_id": "5f9288fd5e25636ce6dcad66",
+                "name": "obs1",
+                "description": "observation1",
+                "solutionId": "5f9288fd5e25636ce6dcad65",
+                "programId": "5d287326652f311044f41dbb"
+            },
+            {
+                "_id": "5fc7aa9e73434430731f6a10",
+                "solutionId": "5fb4fce4c7439a3412ff013b",
+                "programId": "5f5b216a9c70bd2973aee29f",
+                "name": "My Solution",
+                "description": "My Solution Description"
+            }
+        ],
+        "count": 2
+    }}
+    * @apiUse successBody
+    * @apiUse errorBody
+    */
+
+    /**
+      * List of solutions and targetted ones.
+      * @method
+      * @name targetedSolutions
+      * @param {Object} req - request data.
+      * @returns {JSON} List of solutions with targetted ones.
+     */
+
+  async targetedSolutions(req) {
+    return new Promise(async (resolve, reject) => {
+        try {
+
+            let observations = await solutionsHelper.targetedSolutions(
+                req.body,
+                req.query.type,
+                req.userDetails.userInformation.userId,
+                req.pageSize,
+                req.pageNo,
+                req.searchText,
+                req.query.filter,
+                req.query.surveyReportPage ? req.query.surveyReportPage : ""
+            );
+
+            return resolve(observations);
+
+        } catch (error) {
+            return reject({
+                status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+                message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+                errorObject: error
+            });
+        }
+    })
+}
+
+
+
+
+
+
+   /**
+  * @api {post} /project/v1/solutions/verifySolution/:Id
+  * @apiVersion 1.0.0
+  * @apiName verify Solutions targeted
+  * @apiGroup Solutions
+  * @apiSampleRequest /project/v1/solutions/verifySolution/5f6853f293734140ccce90cf
+  * @apiHeader {String} X-authenticated-user-token Authenticity token  
+  * @apiUse successBody
+  * @apiUse errorBody
+  * @apiParamExample {json} Request:
+  * {
+  *   "role" : "HM,DEO",
```

Comment:
@Prajwal17Tunerlabs is the req body is correct after changing to generic scope

Reply:
changed the req body